### PR TITLE
feat(app): clicking the HPI logo opens hpi.de

### LINKED ISSUE #88: Clicking on HPI logo could open browser with hpi.de
**Description**

### PR DESCRIPTION
**Pull request by ctiedt**
**Fixes: #88**

**Checklist**

- [x] Commit messages follow conventionalcommits.org
- [x] Tested on actual devices
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Changes**
The HPI logos on the dashboard and in the bottom drawer are now buttons that lead to hpi.de.
This branch also includes minor changes to the layout of the bottom drawer to make the HPI logo and settings button more centered.

### CODE REVIEW

File: lib/app/widgets/dashboard_page.dart [hunk 1]
```diff
@@ -34,15 +35,20 @@ class DashboardPage extends StatelessWidget {
...
-              child: ConstrainedBox(
-                constraints: BoxConstraints(maxWidth: 200, maxHeight: 100),
-                child: Image.asset('assets/logo/logo_text.png'),
+          FlatButton(
+            child: Padding(
+              padding: const EdgeInsets.symmetric(vertical: 8.0),
+              child: Align(
+                alignment: Alignment.topRight,
+                child: ConstrainedBox(
+                  constraints: BoxConstraints(maxWidth: 200, maxHeight: 100),
```
Comment (MarcelGarus) [inline lib/app/widgets/dashboard_page.dart]: Better use a `LimitedBox` with a `maxWidth` and -`height` here

File: lib/app/widgets/dashboard_page.dart [hunk 2]
```diff
@@ -34,15 +35,20 @@ class DashboardPage extends StatelessWidget {
...
-            child: Align(
-              alignment: Alignment.topRight,
-              child: ConstrainedBox(
-                constraints: BoxConstraints(maxWidth: 200, maxHeight: 100),
-                child: Image.asset('assets/logo/logo_text.png'),
+          FlatButton(
+            child: Padding(
+              padding: const EdgeInsets.symmetric(vertical: 8.0),
+              child: Align(
+                alignment: Alignment.topRight,
```
Comment (MarcelGarus) [inline lib/app/widgets/dashboard_page.dart]: Better merge those into a `Container` with `padding` and `alignment`

File: lib/app/widgets/dashboard_page.dart [hunk 3]
```diff
@@ -34,15 +35,20 @@ class DashboardPage extends StatelessWidget {
...
+                child: ConstrainedBox(
+                  constraints: BoxConstraints(maxWidth: 200, maxHeight: 100),
+                  child: Image.asset('assets/logo/logo_text.png'),
+                ),
               ),
             ),
+            onPressed: () {
+              tryLaunch('https://hpi.de');
+            },
```
Comment (MarcelGarus) [inline lib/app/widgets/dashboard_page.dart]: 🏹 f.

File: lib/app/widgets/main_scaffold.dart [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import 'package:flutter/material.dart' hide Route;
+import 'package:flutter/material.dart' as prefix0;
```
Comment (MarcelGarus) [inline lib/app/widgets/main_scaffold.dart]: Ugh this looks like an artifact of https://github.com/dart-lang/sdk/issues/36890

File: lib/app/widgets/main_scaffold.dart [hunk 5]
```diff
@@ -53,11 +55,18 @@ class MainScaffold extends StatelessWidget {
...
-                SizedBox(width: 16),
-                Image.asset('assets/logo/logo_text.png', height: 56),
+                FlatButton(
+                  child: Padding(
+                      padding: const EdgeInsets.all(8),
+                      child:
+                          Image.asset('assets/logo/logo_text.png', height: 50)),
+                  onPressed: () {
+                    tryLaunch('https://hpi.de');
+                  },
```
Comment (MarcelGarus) [inline lib/app/widgets/main_scaffold.dart]: 🏹

**Review by MarcelGarus — CHANGES_REQUESTED** [review]

**Review by MarcelGarus — APPROVED** [review]

**Review by MarcelGarus — CHANGES_REQUESTED** [review]
* We shouldn't display Material ink wells when clicking on the logo. So, instead of using an `InkWell`, a `GestureDetector` might be sufficient.
* On the dashboard, we shouldn't detect taps on the whole header, just on the logo.
* On the dashboard, the logo should align with the carousel thingies below.

**Comment by phpfs** [comment]
This feature might be confusing if one accidentally activates the button. Maybe an alert with confirmation helps :)

**Comment by MarcelGarus** [comment]
@ctiedt An `AlertDialog` for confirmation would indeed be a great solution.

**Pull request closed without merging**